### PR TITLE
Add memberOf flag output to classes and interfaces. Enabled jump to description in inserted comment.

### PR DESCRIPTION
**Pull request by mredbishop**
Added support for memberOf flag to allow better documentation of class properties and methods.

### CODE REVIEW

**Comment by mredbishop** [comment]
Added a tweak to leave the cursor at the description location of the inserted comment for single inserts.

File: src/documenter.ts [hunk 1]
```diff
@@ -202,6 +213,10 @@ export class Documenter implements vs.Disposable {
                 console.warn("Error in source file update:", error);
             }
         }
+
+        if (goToDescription) setTimeout(() => {
```
Comment (joelday) [inline src/documenter.ts]: Subjective nitpick: use braces here

File: src/documenter.ts [hunk 2]
```diff
@@ -202,6 +213,10 @@ export class Documenter implements vs.Disposable {
...
+            this._goToDescriptionLocation(commentText)
+        }, 100);
```
Comment (joelday) [inline src/documenter.ts]: I wish there was a different way to do this, but if it works, it works. Can you do a quick test of this with a huge ass file?

**Comment by joelday** [comment]
Can you add a setting to disable memberOf output? Would probably be ideal to set it on or off for everything/interfaces/enums. Only interfaces and enums by default?

Thanks a ton for doing all of this!

**Comment by joelday** [comment]
Oh, also, feel free to rev the version, update the readme and credit yourself for contributing.

**Comment by mredbishop** [comment]
No worries, thanks for all the rest of the extension :+1:  

What would you like the options to be exactly for memberOf? We use it most for classes as the JSDoc generator doesn't show the documentation for the class members without it. 

Is there a **good** tool for generating the docs directly from the typescript?

**Comment by mredbishop** [comment]
**As per pull request comments:**

- Added in curly braces as noted in comment. 
- Tested description cursor move on a massive file (4000+ lines, 1000 functions) and it works great. 
- Did a load of other testing using largeFile.ts. 
- Added @memberOf note to the readme.md.
- Added configuration options to turn all this stuff off.
- Updated the version number.
- Added my info to the end of the contributors list.

Ed

**Comment by joelday** [comment]
In my experience, the best bet is to compile to ES6 when documenting, so JSDoc is able to infer from the class declarations itself. It won't pick up enum members though, so those need it either way. I don't think the TypeScript output puts the comments in the right place, unfortunately.

Maybe you can try transforming TypeDoc's raw data to JSDoc's raw data? Then, you wouldn't need any type annotation either.

**Comment by joelday** [comment]
Published!

**Comment by mredbishop** [comment]
Great, thanks Joel. I've installed that from the marketplace now and it's
all working as expected.

Let me know if there are any features bugs that you want looking at. We
have a bit of spare time here and there to work on open source stuff so
might be able to help out.

Cheers,

Ed

**Pull request merged**